src/test/java/org/opensearch/ad/bwc/ADBackwardsCompatibilityIT.java

```diff
@@ -83,25 +131,176 @@ public void testBackwardsCompatibility() throws Exception {
+                    startRealtimeJobForHistoricalDetectorOnNewNode();
+
+                    // Verify all realtime detectors are running realtime job
+                    verifyRealtimeJobRunning();
+                    // Verify realtime and historical task exists for all running realtime detector
+                    verifyAdTasks();
+
+                    // Stop and delete detector
+                    stopAndDeleteDetectors();
                     break;
             }
             break;
         }
     }
 
+    private void verifyAdTasks() throws InterruptedException, IOException {
+        boolean realtimeTaskMissing = false;
+        int i = 0;
+        int maxRetryTimes = 10;
+        do {
+            i++;
+            for (String detectorId : runningRealtimeDetectors) {
+                Map<String, Object> jobAndTask = getDetectorWithJobAndTask(client(), detectorId);
+                AnomalyDetectorJob job = (AnomalyDetectorJob) jobAndTask.get(ANOMALY_DETECTOR_JOB);
+                ADTask historicalTask = (ADTask) jobAndTask.get(HISTORICAL_ANALYSIS_TASK);
+                ADTask realtimeTask = (ADTask) jobAndTask.get(REALTIME_TASK);
+                assertTrue(job.isEnabled());
+                assertNotNull(historicalTask);
+                if (realtimeTask == null) {
+                    realtimeTaskMissing = true;
+                }
+                if (i >= maxRetryTimes) {
+                    assertNotNull(realtimeTask);
+                    assertFalse(realtimeTask.isDone());
+                }
+            }
+            if (realtimeTaskMissing) {
+                Thread.sleep(10_000);// sleep 10 seconds to wait for realtime job to backfill realtime task
+            }
+        } while (realtimeTaskMissing && i < maxRetryTimes);
+    }
+
+    private void stopAndDeleteDetectors() throws Exception {
+        for (String detectorId : runningRealtimeDetectors) {
+            deleteRunningDetector(detectorId);
+            Response stopRealtimeJobResponse = stopRealtimeJob(client(), detectorId);
+            assertEquals(RestStatus.OK, TestHelpers.restStatus(stopRealtimeJobResponse));
+            try {
+                Response stopHistoricalAnalysisResponse = stopHistoricalAnalysis(client(), detectorId);
+                assertEquals(RestStatus.OK, TestHelpers.restStatus(stopHistoricalAnalysisResponse));
+            } catch (Exception e) {
+                if (!ExceptionUtil.getErrorMessage(e).contains("No running task found")) {
+                    throw e;
+                }
+            }
+            Map<String, Object> jobAndTask = getDetectorWithJobAndTask(client(), detectorId);
+            AnomalyDetectorJob job = (AnomalyDetectorJob) jobAndTask.get(ANOMALY_DETECTOR_JOB);
+            ADTask historicalAdTask = (ADTask) jobAndTask.get(HISTORICAL_ANALYSIS_TASK);
+            if (!job.isEnabled() && historicalAdTask.isDone()) {
```

Comment:
If you're already asserting that the responses from stopping the RT and historical task are true, should there be asserts that the job is disabled and the historical task is done as well?

Reply:
Historical task stopping is try the best, not 100% sure. Add assertion for historical task state will make the test flaky.

Realtime job stopping should be disabled once the request returns, but actually it's flaky as the index refresh needs some time. Will test if sleep for some time can fix. But from previous experience, sometimes the flaky test case can pass on dev host but will fail in Github workflow. That's why I just assert the stop request returns OK.

Reply:
Got it, makes sense, since it relies on the index updates. I'm ok with leaving as-is.